lib/src/rbac/default_role_manager.dart

```diff
@@ -60,85 +128,111 @@ class DefaultRoleManager implements RoleManager {
   /// aka role: [name1] does not inherit role: [name2] any more.
   /// [domain] is a prefix to the roles.
   @override
-  void deleteLink(String name1, String name2, [String? domain]) {
-    if (domain != null) {
-      name1 = domain + '::' + name1;
-      name2 = domain + '::' + name2;
+  void deleteLink(String name1, String name2,
+      [List<String> domain = const []]) {
+    if (domain.isEmpty) {
+      domain = [DEFAULT_DOMAIN];
+    } else if (domain.length > 1) {
+      throw Exception('error: domain should be 1 parameter');
     }
 
-    if (!hasRole(name1) || !hasRole(name2)) {
+    final allRoles = loadOrDefault(allDomains, domain[0], DomainRoles());
+
+    if (!allRoles.roles.containsKey(name1) ||
+        !allRoles.roles.containsKey(name2)) {
       throw ArgumentError('error: name1 or name2 does not exist');
     }
```

Comment:
Maybe check each parameter on its own so we can let know the developer which parameter, if any, doesn't exist.